### PR TITLE
Reorganize exceptions into domain packages

### PR DESCRIPTION
**Pull request by mattnichols**
# Summary of Changes

* Placing exceptions in packages along side related classes

## Public API Additions/Changes

New exception classes. Deprecated MdxApiException

## Downstream Consumer Impact

New classes only. No impact.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: common/src/main/java/com/mx/common/connect/ConnectException.java [hunk 1]
```diff
@@ -1,6 +1,8 @@
-package com.mx.common.exception;
+package com.mx.common.connect;
 
+import com.mx.common.accessors.AccessorException;
```
Comment (stevecl5) [inline common/src/main/java/com/mx/common/connect/ConnectException.java:3]: Do we still want this to extend `AccessorException`, or were we going to change it to inherit from `PathRequestException` directly?

File: common/src/main/java/com/mx/common/exception/PathRequestException.java [hunk 2]
```diff
@@ -6,7 +6,22 @@
...
+import com.mx.common.connect.ServiceUnavailableException;
+import com.mx.common.connect.TimeoutException;
+import com.mx.common.connect.TooManyRequestsException;
+import com.mx.common.connect.UpstreamErrorException;
+import com.mx.common.facility.FacilityException;
+import com.mx.common.gateway.BehaviorException;
+import com.mx.common.gateway.GatewayException;
 
 /**
  * Base exception for any request-based error. These are errors that occur at any level during a request.
```
Comment (stevecl5) [inline common/src/main/java/com/mx/common/exception/PathRequestException.java:27]: If we change `ConnectException` to extend `PathRequestException` the structure will need to be updated here as well (in the Javadoc).

File: common/src/main/java/com/mx/common/connect/ConnectException.java (hunk 1 above)
Comment (mattnichols) [reply common/src/main/java/com/mx/common/connect/ConnectException.java]: Fixed

File: common/src/main/java/com/mx/common/exception/PathRequestException.java (hunk 2 above)
Comment (mattnichols) [reply common/src/main/java/com/mx/common/exception/PathRequestException.java:27]: Fixed

**Review by stevecl5 — APPROVED** [review]

**Pull request merged**